src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/ProjectFile.cs

```diff
@@ -217,7 +218,24 @@ private DocumentFileInfo MakeNonSourceFileDocumentFileInfo(MSB.Framework.ITaskIt
             var isLinked = IsDocumentLinked(documentItem);
             var isGenerated = IsDocumentGenerated(documentItem);
 
-            return new DocumentFileInfo(filePath, logicalPath, isLinked, isGenerated, SourceCodeKind.Regular);
+            var folders = GetRelativeFolders(documentItem);
+            return new DocumentFileInfo(filePath, logicalPath, isLinked, isGenerated, SourceCodeKind.Regular, folders);
+        }
+
+        private ImmutableArray<string> GetRelativeFolders(MSB.Framework.ITaskItem documentItem)
+        {
+            var linkPath = documentItem.GetMetadata(MetadataNames.Link);
+            if (!RoslynString.IsNullOrEmpty(linkPath))
+            {
+                return linkPath.Split(PathUtilities.DirectorySeparatorChar).ToImmutableArray();
```

Comment:
Do we also need to split on the AltDirectorySeperatorChar too? I have no idea what the behavior is with a forward slash for our existing project systems.

Reply:
So does the link path here contain the file name too? Or did the GetMetadata() somehow already filter that out? It looks like your test is passing but it's not entirely clear to me why.

Reply:
> Do we also need to split on the AltDirectorySeperatorChar too? I have no idea what the behavior is with a forward slash for our existing project systems.

Emm, maybe the file name is filtered out in later code. It should have PathUtilities.GetDirectoryName()

>

Reply:
> Do we also need to split on the AltDirectorySeperatorChar too? I have no idea what the behavior is with a forward slash for our existing project systems.

Added.